core/include/jiminy/core/robot/Robot.h

```diff
@@ -65,6 +66,38 @@ namespace jiminy
                             Eigen::Ref<vectorN_t const> const & v,
                             Eigen::Ref<vectorN_t const> const & a,
                             vectorN_t                   const & u);
+
+        /// \brief Add a kinematic constraint to the robot.
+        ///
+        /// \param[in] constraintName Unique name identifying the kinematic constraint.
+        /// \param[in] constraint Constraint to add.
+        hresult_t addConstraint(std::string const & constraintName,
```

Comment:
I think it would be at Model level instead of Robot.

Reply:
I placed it at Robot level because it currently behaves like a motor / sensor (i.e. an extral object one can attach / detach, not an intrinsic property of the model). So it's linked to the previous debate.

Reply:
I see your point but even though the mechanism is the same, I don't fully agree that it is more related to the Robot that Model. One could use a Model instead of a Robot if the system does not feature "physical" plug-ins such as motors or sensors. But It should still be possible to add wheel constraint. For example to simulate a simple ball rolling on the ground.